### ISSUE TITLE
The crash dialog box is intermittently displayed after quitting from 360 videos and resuming the app

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [MozillaReality/FirefoxReality#1959]
**[Affected Versions]:**
- Firefox Reality version 5-32821846 RC1 (58b59be3)

**[Affected Platforms]:**
- Oculus Quest - Android 7.1.1
- HTC Vive Focus Plus - Android 7.1.2
- Oculus Go - Android 7.1.2

**[Prerequisites]:**
- The headset is turned on and the browser is opened.

**[Steps to reproduce]:**
1. Open and Play a 360 video in immersive mode.
2. Click the "Oculus/Vive" button from the controller.
3. Click the "Quit/Home" button from the prompt displayed.
4. Resume the app and observe the behavior.

**[Expected results]:**
- The app is correctly resumed.

**[Actual results]:**
- The crash dialog box is intermittently displayed.

**[Additional Notes]:**
- This issue is not reproducible on the HTC Vive Focus and Lenovo Mirage Solo devices.
- Attached are the log cat, crash report  and a screed recording of the issue:
	- Logcat: [link](https://drive.google.com/open?id=1nEjl42r3Xlr50pwXDjs9_avGlHM75odQ)
	- Crash Report: [link](https://crash-stats.mozilla.com/report/index/bp-416d8ec1-66dd-4deb-8e9f-041c90191011)
	- Screen recording: [link](https://drive.google.com/open?id=1Shr7VZ6fYOJV5lSSeza1Uu1c33xWCtMQ)

**Comment by jvonitter** [comment]
The crash here is caused by a known Oculus bug, so we can't fix it right now. We could suppress the crash reports, but I would like to hold off doing that for now so we can collect the data. Since this only happens if you quit the app while in video immersive mode, the impact shouldn't be too bad.

I'm going to push this to 7 to revisit then.

**Comment by Softvision-RemusDranca** [comment]
@jvonitter This issue is also reproducible on HTC Vive Focus Plus.  I’m not sure if this will change the milestone set for it after taking this into consideration.
The attachments above are recorded with the  HTC Vive Focus Plus device. I should have added the name of the device in front of them, sorry for misleading you on this.

**Comment by bluemarvin** [comment]
It's a bug in Android 7 which is unfortunately the version that most of the supported headsets are currently running. This is the same issue documented in #1770. I changed the way crashes are handled to prevent FxR from losing crash reports so now the user will be prompted with a crash dialog the next time they start FxR if it crashed on exit.

**Comment by Softvision-RemusDranca** [comment]
@bluemarvin I've retested this scenario with the latest 6-32902155 (93816a5a) build and it seems that after using the app for ~ one hour the issue is continuously reproducible.
Also, The HTC devices have ~ one minute where the screen is black until the app is resumed and the crash dialog is displayed and intermittently closes instead of resuming.
I've attached the videos and logcats tested with this build and I want you to confirm, if possible, that the same bug in Android 7 is causing this behavior or that I should log a new issue. Thanks in advance!
- HTC Vive Focus Plus screen recording: [link](https://drive.google.com/open?id=12sVcUewWQ58JyOEfKN3D6j6GM9YOUOrc)
- HTC Vive Focus Plus logcat: [link](https://drive.google.com/open?id=1BOUV8mmZEd8ld-F-JhJf5vngxLiows6M)
- HTC Vive Focus Plus crash report: [link](https://crash-stats.mozilla.com/report/index/bp-51b4ebe4-812c-4d84-a04b-fbb250191018)
- Oculus Quest logcat: [link](https://drive.google.com/open?id=1miDAl4EzUCZTAp14EEC_pkpYkVnIaXHj)
- Oculus Quest crash report: [link](https://drive.google.com/open?id=1NGfYxk-Rzv8gU_uvvJUPVqdoGZFZDzYZ)

**Comment by jvonitter** [comment]
@Softvision-RemusDranca Can you check this one again? It may have been fixed by some other changes we did last week.

**Comment by Softvision-GeluHaiduc** [comment]
I managed to partially reproduce the issue using Firefox Reality version 7 RC3 (7b66fb14) installed on HTC Vive Focus Plus.
After more than 1 hour of stressing the app navigating Youtube and quitting and resuming the app creating tabs and windows, the app crashed, the logcat file is [here](https://drive.google.com/open?id=1nMt0Z6kDQ-cO-he5YT3Nlln6L7hgTpA1) and the crash report [here](https://crash-stats.mozilla.com/report/index/0451006b-4be4-47c7-97e9-749950191127).

**Comment by jvonitter** [comment]
I think if it takes an hour and a lot of work to repro it, we can downgrade this from Major to Normal. I'll leave it open so we can investigate further.

**Comment by MortimerGoro** [comment]
The latest crash report is different from the original one. We have already landed a fix for this crashlog (https://github.com/MozillaReality/FirefoxReality/pull/2386)

If we cant reproduce the original one (https://crash-stats.mozilla.com/report/index/416d8ec1-66dd-4deb-8e9f-041c90191011) we could close this.

**Comment by jvonitter** [comment]
Thanks @MortimerGoro! @Softvision-GeluHaiduc let us know if you can repro the original crash. If not, go ahead and close this issue.

**Comment by MortimerGoro** [comment]
Fixed in #2482

**Comment by Softvision-RemusDranca** [comment]
I have verified this issue and it's no longer reproducible with the latest Firefox Realty 8 RC2 build (4ff1f59). I was able to reproduce this behavior only after stressing the app for a long period of time but not after resuming from 360 videos and I wasn't sure that the crashes are related so I opened a new issue, #2663, for tracking purpose. 
Also I encountered issue #2675, which could block us from reproducing this issue but since this issue is intermittent it's save to close in as verified fixed and log a new one if encountered in the future.

Tested on Oculus Go (Android 7.1.2), Oculus Quest (Android 7.1.1), HTC Vive Focus Plus (Android 7.1.2) and HTC Vive Focus (Android 7.1.2).
